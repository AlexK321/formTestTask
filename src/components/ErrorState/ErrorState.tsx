import React, { FC } from 'react';
import './ErrorState.less';

interface ParamTypes {
  error?: string;
}

const ErrorState: FC<ParamTypes> = ({ error }) => {
  return (
    <div className="error-wrapper">
      <div className="error-block">
        <h2>Что-то пошло не так</h2>
        {error && <p>{error}</p>}
      </div>
    </div>
  );
};

export default ErrorState;
