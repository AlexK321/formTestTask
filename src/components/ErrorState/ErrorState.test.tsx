import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ErrorState from './ErrorState';

interface ParamTypes {
  error: string;
}

describe('Component: ErrorState', () => {
  const setUp = (props?: ParamTypes) => shallow(<ErrorState {...props} />);
  let component: ShallowWrapper;

  beforeEach(() => {
    component = setUp();
  });

  it('match snapshot component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render error text', () => {
    const errorText = 'error test text';

    component = shallow(<ErrorState error={errorText} />);
    const errorBlock = component.find('p');

    expect(errorBlock.text()).toBe(errorText);
  });
});
