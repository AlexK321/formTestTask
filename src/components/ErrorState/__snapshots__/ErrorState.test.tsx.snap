// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorState match snapshot component 1`] = `
<div
  className="error-block"
>
  <h2>
    Что-то пошло не так
  </h2>
</div>
`;

exports[`Component: SubmitResult match snapshot component 1`] = `
<div
  className="error-block"
>
  <h2>
    Что-то пошло не так
  </h2>
</div>
`;
